tests/unit/src/components/shared/AppFooter.spec.js

```diff
@@ -0,0 +1,142 @@
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  beforeAll(() => {
+    Object.defineProperty(window, 'localStorage', {
+      value: {
+        getItem: jest.fn(),
+        writable: true
+      }
+    })
+  });
+
+  describe('data', () => {
+    it('should have "isBackToTopVisible" as false value', () => {
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isBackToTopVisible).toEqual(false);
+    });
+
+    it('should have "isFooterSubscribeVisible" as true value if "subscribeClosed" value is falsey', () => {
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isFooterSubscribeVisible).toEqual(true);
+    });
+
+    it('should have "isFooterSubscribeVisible" as true value if "subscribeClosed" value is falsey', () => {
+      localStorage.getItem.mockReturnValue(false);
+      
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isFooterSubscribeVisible).toEqual(true);
+    });
+
+    it('should have "isFooterSubscribeVisible" as false value if "subscribeClosed" value is true', () => {
+      localStorage.getItem.mockReturnValue(true);
+      
+      const componentState = AppFooter.data();
+      
+      expect(componentState.isFooterSubscribeVisible).toEqual(false);
+    });
+  });
+
+  describe('methods', () => {
+    describe('handleScroll', () => {
+      const setScrollY = (value) => {
+        Object.defineProperty(window, 'scrollY', {
+          value: value,
+          writable: true
+        });
+      };
+
+      it('assign "isBackToTopVisible" to true, if scrollY is greater than 540', () => {
+        const wrapper = shallowMount(AppFooter, {
+          stubs: ['router-link'],
+        });
+
+        setScrollY(541);
+
+        wrapper.vm.handleScroll();
+
+        expect(wrapper.vm.isBackToTopVisible).toEqual(true);
+      });
+
+      it('assign "isBackToTopVisible" to false, if scrollY is smaller than 540', () => {
+        const wrapper = shallowMount(AppFooter, {
+          stubs: ['router-link'],
+        });
+
+        setScrollY(539);
+
+        wrapper.vm.handleScroll();
+
+        expect(wrapper.vm.isBackToTopVisible).toEqual(false);
+      });
+
+      it('assign "isBackToTopVisible" to false, if scrollY is equal than 540', () => {
+        const wrapper = shallowMount(AppFooter, {
+          stubs: ['router-link'],
+        });
+
+        setScrollY(540);
+
+        wrapper.vm.handleScroll();
+
+        expect(wrapper.vm.isBackToTopVisible).toEqual(false);
+      });
+    });
+  });
+
+  describe('templates', () => {
+    const shallowAppFooter = () => {
+      return shallowMount(AppFooter, {
+        stubs: ['router-link'],
+      });
+    }
+
+    it('should match the snapshot', () => {
+      const wrapper = shallowAppFooter();
+
+      expect(wrapper.vm.$el).toMatchSnapshot();
+    });
+
+    it('should not render .has-widget div which its id is #footer, if subscribeClosed is true', () => {
```

Comment:
I prefer `js` prefix while using DOM elements on js side. It would be better. WDYT @fatihacet @mersanuzun  ?

Reply:
I don't understand the issue :)

Reply:
There is no issue :) It is a recommendation.